### PR TITLE
Coupon Changes

### PR DESCRIPTION
**Pull request by DeepaliAgrwl**

### CODE REVIEW

**Review by rishabh-behl — CHANGES_REQUESTED** [review]
@DeepaliAgrwl : import com.hcl.ecomm.core.services.CouponService is Not checked in .. Build is failing

File: core/src/main/java/com/hcl/ecomm/core/servlets/CouponServlet.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            property = {"sling.servlet.paths=/bin/hclecomm/applyCoupon",
+                    "sling.servlet.method=" + HttpConstants.METHOD_GET,
+                    "sling.servlet.extensions=json"})
+
+public class CouponServlet extends SlingSafeMethodsServlet {
+
+    private static final long serialVersionUID = 4016057296495129474L;
+    private static final Logger LOG = LoggerFactory.getLogger("CouponServlet.class");
+
+    String COUPON_LIST_PATH = "/etc/acs-commons/lists/coupon-list";
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/servlets/CouponServlet.java:33]: @DeepaliAgrwl : /etc/acs-commons/lists/coupon-list needs to be checked in as part of codebase with mode = update

File: core/src/main/java/com/hcl/ecomm/core/servlets/CouponServlet.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            Resource child = children.next();
+            ValueMap childNodeProperty = child.getValueMap();
+            String couponCode = (String)childNodeProperty.get("jcr:title");
+            if(couponApplied.equals(couponCode))
+            {
+                responseString = (String)childNodeProperty.get("value");
+                break;
+            }
+        }
+        response.getWriter().write(responseString);
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/servlets/CouponServlet.java]: update as discount price

File: core/src/main/java/com/hcl/ecomm/core/servlets/CouponServlet.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response) throws ServletException, IOException {
+        String couponApplied = request.getParameter("coupon");
+        String responseString = "";
+        ResourceResolver resourceResolver = request.getResourceResolver();
+        PageManager pageManager = resourceResolver.adaptTo(PageManager.class);
+        Page listPage = pageManager.getPage(COUPON_LIST_PATH);
+        Resource resource = listPage.adaptTo(Resource.class);
+        Resource childResources = resource.getChild("jcr:content").getChild("list");
+
+        Iterator<Resource> children = childResources.listChildren();
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/servlets/CouponServlet.java]: update to coupon nodes

File: ui.apps/src/main/content/jcr_root/etc/acs-commons/lists/.content.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="sling:OrderedFolder">
+    <font-awesome-icons/>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/etc/acs-commons/lists/.content.xml]: @DeepaliAgrwl : why do we have other list here ?

File: core/src/main/java/com/hcl/ecomm/core/servlets/CouponServlet.java (hunk 1 above)
Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/servlets/CouponServlet.java:33]: done.

File: core/src/main/java/com/hcl/ecomm/core/servlets/CouponServlet.java (hunk 2 above)
Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/servlets/CouponServlet.java]: Variable name has been changed

File: core/src/main/java/com/hcl/ecomm/core/servlets/CouponServlet.java (hunk 3 above)
Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/servlets/CouponServlet.java]: Node variable names has been changed

File: ui.apps/src/main/content/jcr_root/etc/acs-commons/lists/.content.xml (hunk 4 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/etc/acs-commons/lists/.content.xml]: removed the extra lists from code.

**Review by rishabh-behl — APPROVED** [review]
LGTM

**Pull request merged**